### PR TITLE
Fix rowDescription parsing, typeParsers, timezone tests, and prepared statement tests

### PR DESCRIPTION
**Pull request by DMickens**
This may have a significant impact on the state of some tests. Anything that was modified to expect strings for primitives in the result rows will need to be updated. This will depend on how loosely results were compared to be equal. This should also resolve some of the issues identified on the confluence page for documented test failures related to parsing unexpected typeIDs. Once other current PRs are merged, I will need to merge master back into this branch and retest for regressions. Some regressions may be unavoidable, but still need documented

### CODE REVIEW

**Comment by DMickens** [comment]
This has expanded in scope to also include fixing huge-numeric-tests, no-row-result-tests.js, and type-parser-override-tests.js as it took only a small step from this PR to fixing those tests. Those are all associated with VER-82555

File: packages/vertica-nodejs/test/integration/client/timezone-tests.js [hunk 1]
```diff
@@ -10,22 +10,29 @@ var date = new Date()
 const pool = new helper.vertica.Pool()
 const suite = new helper.Suite()
 
+/*
+* NOTE: These tests right now are not very useful. We are not parsing timestamps into date objects,
```
Comment (jonathanl-bq) [inline packages/vertica-nodejs/test/integration/client/timezone-tests.js:14]: Does it make sense to create a JIRA ticket for this?

File: packages/vertica-nodejs/lib/type-overrides.js [hunk 2]
```diff
@@ -20,6 +20,22 @@ function TypeOverrides(userTypes) {
   this._types = userTypes || types
   this.text = {}
   this.binary = {}
+  // this is a 'temporary' solution allowing us to continue to use pg-types as long as we can to avoid another large
```
Comment (jonathanl-bq) [inline packages/vertica-nodejs/lib/type-overrides.js:23]: We should create a task for creating a v-types package, if it hasn't been created yet.

File: packages/vertica-nodejs/test/integration/client/huge-numeric-tests.js [hunk 3]
```diff
@@ -5,15 +5,15 @@ const pool = new helper.vertica.Pool()
...
+    types.setTypeParser(16, function () {
       return 'yes'
     })
-    types.setTypeParser(1700, 'binary', function () {
+    types.setTypeParser(16, 'binary', function () { // irrelevant until we support binary transfer
       return 'yes'
     })
     var bignum = '294733346389144765940638005275322203805'
-    client.query('CREATE LOCAL TEMP TABLE bignumz(id numeric(64,0))')
+    client.query('CREATE LOCAL TEMP TABLE bignumz(id numeric(40,0))')
```
Comment (DMickens) [inline packages/vertica-nodejs/test/integration/client/huge-numeric-tests.js:16]: no significant difference, just result of merge conflict

**Review by jonathanl-bq — APPROVED** [review]
Approved. I left a couple of minor comments, but not blocking merge.

File: packages/vertica-nodejs/test/integration/client/timezone-tests.js (hunk 1 above)
Comment (DMickens) [reply packages/vertica-nodejs/test/integration/client/timezone-tests.js:14]: I have a jira for long term type parsing fix, I think it may make sense to add this as a comment onto that jira

File: packages/vertica-nodejs/lib/type-overrides.js (hunk 2 above)
Comment (DMickens) [reply packages/vertica-nodejs/lib/type-overrides.js:23]: we have one

File: packages/v-protocol/src/parser.ts [hunk 4]
```diff
@@ -291,8 +291,12 @@ export class Parser {
   private parseField(): Field {
     const name = this.reader.cstring()
     const tableID = this.reader.uint64()
-    const schemaName = this.reader.cstring()
-    const tableName = this.reader.cstring()
+    var schemaName = ""
```
Comment (alexr-bq) [inline packages/v-protocol/src/parser.ts:294]: Was this resulting in the packet reading getting mangled in some case (invalid tableID?)

Comment (DMickens) [reply packages/v-protocol/src/parser.ts:294]: yes exactly. Wasn't playing nicely with queries that were returning literals, which would get resolved later in the parsing, but caused a few metadata fields like typeID to be wrong

File: packages/vertica-nodejs/test/integration/client/no-data-tests.js [hunk 5]
```diff
@@ -13,7 +13,7 @@ suite.test('noData message handling', function () {
 
   var q = client.query({
     name: 'boom',
-    text: 'create table if not exists boom(id identity, size integer)',
+    text: 'create table if not exists boom(size integer)',
```
Comment (jonathanl-bq) [inline packages/vertica-nodejs/test/integration/client/no-data-tests.js:16]: Can we switch this to a local temp table?

**Review by jonathanl-bq — APPROVED** [review]
One small comment. Otherwise, looks good!

File: packages/vertica-nodejs/test/integration/client/no-data-tests.js (hunk 5 above)
Comment (DMickens) [reply packages/vertica-nodejs/test/integration/client/no-data-tests.js:16]: yep, thats part of the reason why I removed identity, I just forgot. Making that change

File: packages/vertica-nodejs/test/integration/client/type-parser-override-tests.js [hunk 6]
```diff
@@ -20,7 +19,7 @@ pool.connect(
   assert.success(function (client1, done1) {
     pool.connect(
       assert.success(function (client2, done2) {
-        var boolTypeOID = VerticaType.boolTypeOID
```
Comment (williammatherjones) [inline packages/vertica-nodejs/test/integration/client/type-parser-override-tests.js:23]: Why did you remove the common type definitions?   Referring to values using the hard coded number is not as good of a practice than using a common definition.

Comment (DMickens) [reply packages/vertica-nodejs/test/integration/client/type-parser-override-tests.js:23]: This wasn't working. It was either imported incorrectly or exported incorrectly in v-protocol, but boolTypeOID was being given a value of undefined, so the type parser for booleans wasn't being used. 

I agree that we need to refer to VerticaType when we can and I have already logged a jira for making these changes on a larger scale throughout the driver and driver tests.

**Review by williammatherjones — APPROVED** [review]
